lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -119,10 +134,13 @@ public void addLogicalTypeConversion(Conversion<?> conversion) {
   @SuppressWarnings("unchecked")
   public <T> Conversion<? extends T> getConversionTo(Class<T> datumClass,
                                                      LogicalType logicalType) {
-    Conversion<?> conversion = conversionsByClass.get(datumClass);
-    if (conversion != null &&
-        conversion.getLogicalTypeName().equals(logicalType.getName())) {
-      return (Conversion<T>) conversion;
+    List<Conversion<?>> conversions = conversionsByClass.get(datumClass);
+    if (conversions != null) {
```

Comment:
My mistake, this is from the original patch evolving to meet the performance requirements. This should be a single method now.